documentation/query-api.md

```diff
@@ -0,0 +1,213 @@
+
+    {<service-name>: {
+        "service_version": <service-version>,
+        "service_status_version": <service-status-version>,
+        "detail_level": <detail-level>,
+        "is_running": <service-state>,
+        "status": <any>
+        },
+     <service-name>: {
+        ...
+        },
+    ...
+    }
+
+For detailed information, please see the [Wire Format Specification](./wire-formats.md).
+
+NOTE: If any services in your application have registered more than one supported
+status format version, and you are querying at a detail level other than `critical`
+this endpoint will *always* return the latest format.  Therefore,
+if you need to ensure backward compatibility across upgrades of your application,
+you should consider using the [`/services/<service-name>`](#GETStatusV1ServiceServiceName) endpoint (which
+returns status info for a single service), rather than the `/services` endpoint
+(which aggregates status for all registered services).
+
+### Examples
+
+Using `curl` from localhost:
+
+Get the service status of all registered services in an application:
+
+    curl -k https://localhost:8000/status/v1/services
+
+    {
+        "puppet-server": {
+            "detail_level": "info",
+            "is_running": "true",
+            "service_status_version": 1,
+            "service_version": "1.0.9-SNAPSHOT",
+            "status": {
+                "bar": "bar",
+                "foo": "foo"
+            }
+        },
+
+        "other-service": {
+            "detail_level": "info",
+            "is_running": "true",
+            "service_status_version": 2,
+            "service_version": "0.0.1-SNAPSHOT",
+            "status": {
+                "baz": [1, 2, 3],
+                "bang": {"key": "value"}
+            }
+        }
+    }
+
+Get the service status of all registered services in an application, at a
+ specified level of detail:
+
+    curl -k "https://localhost:8140/status/v1/services?level=critical"
+
+    {
+        "puppet-server": {
+            "detail_level": "critical",
+            "is_running": "true",
+            "service_status_version": 1,
+            "service_version": "1.0.9-SNAPSHOT",
+            "status": null
+        },
+
+        "other-service": {
+            "detail_level": "info",
+            "is_running": "true",
+            "service_status_version": 2,
+            "service_version": "0.0.1-SNAPSHOT",
+            "status": null
+        }
+    }
+
+## `GET /status/v1/services/<service-name>`
+
+This will return status information for a single, specified service from the running
+application.
+
+### URL Parameters
+
+* `level`: Optional.  A JSON String from among the legal
+[Status Detail Levels](#StatusDetailLevel) listed above.  Status information for
+the requested service will be provided at the requested level of detail.  If
+not provided, the default level is `"info"`.
+
+* `service-status-version`: Optional.  A JSON integer specifying the desired status
```

Comment:
This should be `service_status_version` I think, because we're using underscores in query params.